sentry_sdk/integrations/gql.py

```diff
@@ -0,0 +1,90 @@
+import gql  # type: ignore[import]
+from graphql import print_ast, get_operation_ast, DocumentNode, VariableDefinitionNode  # type: ignore[import]
+from gql.transport import Transport, AsyncTransport  # type: ignore[import]
+from gql.transport.exceptions import TransportQueryError  # type: ignore[import]
+from sentry_sdk.utils import event_from_exception
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations import Integration
+from typing import Any, Dict, Optional, Tuple, Union
```

Comment:
The usual pattern in the SDK is to hide everything imported only for the sake of type hints inside an `if TYPE_CHECKING:` check, I'm assuming to avoid importing a lot of stuff that isn't actually used. For integrations that use old-style type hints this is straightforward (just import the things in the `if` and use them in type comments), but for new-style hints where the hints are actual code this would probably not work. This can be circumvented by using strings instead of the objects themselves, see [e.g. the starlite integration](https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/integrations/starlite.py).

I'd suggest we either turn the type hints into type comments (preferred from my POV to be consistent with the rest of the SDK -- the starlite integration is an outlier) or we do it the starlite way.

Reply:
Yeah, using `if TYPE_CHECKING` with new type hints is not super-straightforward. I was unaware of the possibility of using strings the way the starlite integration does, so if we stay with the new-style hints we should probably switch to strings.

Another [possibility](https://peps.python.org/pep-0563/) which works in Python 3.7+ is to simply add `from __future__ import __annotations__` to the start of the file; then all type annotations are automatically converted to strings behind the scenes and will not be evaluated at runtime, allowing us to use `if TYPE_CHECKING` the same way we currently do in most of the codebase. 


> I'd suggest we either turn the type hints into type comments (preferred from my POV to be consistent with the rest of the SDK -- the starlite integration is an outlier) or we do it the starlite way.

The reason I went with type hints instead of type comments is because it was my understanding that if we were ever to drop support for Python 2.7, we would likely want to change our code to use type hints instead of type comments. Since this file is only used in Python 3.7+, I thought it would make sense to write it with type hints so we would have fewer things to change if and when we drop Python 2.7 support. Although I guess it would probably also be possible to use an automated refactoring tool to change all our type comments into type hints, so perhaps this is not something to seriously worry about.